### ISSUE TITLE
About pose optimizer 

### ISSUE THREAD

**Issue by jinmaodaomaye2021** [HengyiWang/Co-SLAM#27]
Hi,

Thanks for your great work. I looked at your code and found the implementation uses a shared Adam optimizer for all poses in global BA.

My question is that if some poses are not selected in a global BA iteration, do they get updated in the optimizer ? 

- My expectation is that if the randomly selected rays don't have pose A, pose A shouldn't be updated.
- However, the Adam optimizer may use history gradient to update all parameters even if some parameters have zero gradient.

Do you think this is a valid concern ?
Thanks.

**Comment by HengyiWang** [comment]
Hi @jinmaodaomaye2021, thank you for your question.

It seems you are right, this is a valid concern. The Adam optimizer uses the history gradient to update all parameters. Currently, for each global BA, we initialize a new optimizer and update the pose parameters only twice, so the impact of this will not be significant. However, I will try to do some experiments to see if this will indeed have a negative impact on pose optimization when I have time. If so, I will switch to [SparseAdam](https://pytorch.org/docs/stable/generated/torch.optim.SparseAdam.html#torch.optim.SparseAdam) for global BA.
